osu.Game/Overlays/BeatmapListing/BeatmapListingSortTabControl.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Bindables;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Graphics;
+using osuTK.Graphics;
+using osuTK;
+using osu.Framework.Input.Events;
+
+namespace osu.Game.Overlays.BeatmapListing
+{
+    public class BeatmapListingSortTabControl : OverlaySortTabControl<BeatmapSortCriteria>
+    {
+        public readonly Bindable<SortDirection> SortDirection = new Bindable<SortDirection>(Overlays.SortDirection.Descending);
+
+        public BeatmapListingSortTabControl()
+        {
+            Current.Value = BeatmapSortCriteria.Ranked;
+        }
+
+        protected override SortTabControl CreateControl() => new BeatmapSortTabControl
+        {
+            SortDirection = { BindTarget = SortDirection }
+        };
+
+        private class BeatmapSortTabControl : SortTabControl
+        {
+            public readonly Bindable<SortDirection> SortDirection = new Bindable<SortDirection>();
+
+            protected override TabItem<BeatmapSortCriteria> CreateTabItem(BeatmapSortCriteria value) => new BeatmapSortTabItem(value)
+            {
+                SortDirection = { BindTarget = SortDirection }
+            };
+
+            private class BeatmapSortTabItem : SortTabItem
+            {
+                public readonly Bindable<SortDirection> SortDirection = new Bindable<SortDirection>();
+
+                public BeatmapSortTabItem(BeatmapSortCriteria value)
+                    : base(value)
+                {
+                }
+
+                protected override TabButton CreateTabButton(BeatmapSortCriteria value) => new BeatmapTabButton(value)
+                {
+                    Active = { BindTarget = Active },
+                    SortDirection = { BindTarget = SortDirection }
+                };
+
+                private class BeatmapTabButton : TabButton
```

Comment:
Same thing with the nesting here, this doesn't need to be encapsulated this much.